### PR TITLE
Arr.ai shell tutorial

### PR DESCRIPTION
**Pull request by marcelocantos**
Checklist:
- [n/a] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

File: docs/tutorial/shell.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
+# Arr.ai shell basics
+
+This tutorial will provide a brief walkthrough of the features of arr.ai's
+interactive shell.
+
+## Starting the shell
+
+There are two ways to start the interactive. The first is to use the `arrai i`
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:8]:
```suggestion
There are two ways to start the interactive shell. The first is to use the `arrai i`
```

File: docs/tutorial/shell.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+```arrai
+@> (function1(a, b, c)
+ >  + function2(d, e, f)
+ >  + function3(x, y, z)
+ > ) / 3
+```
+
+Observe that the prompt changes from `@>` to `>` to indicate that it's awaiting
+further input. Here's a more cmomplex example.
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:63]:
```suggestion
further input. Here's a more complex example.
```

File: docs/tutorial/shell.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+```arrai
+@> /set x = 42
+42
+@> /set y = 54
+54
+@> x / y
+0.7777777777777778
+```
+
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:95]: Perhaps something like 

> For more details refer to [Set](https://github.com/arr-ai/arrai/blob/master/README.md#set)

File: docs/tutorial/shell.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+```arrai
+@> x / y
+0.7777777777777778
+@> /unset x
+@> x / y # FAIL
+@> 1 / y
+0.018518518518518517
+```
+
+You may have noticed above that arr.ai has something looking like an assignment
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:107]:
```suggestion
You may have noticed above that arr.ai has something that looks like an assignment
```

File: docs/tutorial/shell.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+statement: `let NAME = EXPR;`. You might then be wondering why we even need
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:108]:
```suggestion
statement: `let NAME = EXPR;`. You may then be wondering why we even need
```

File: docs/tutorial/shell.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+`/set`.
+
+This form is known as a let-binding and it is *not* in fact a statement. The
+full form is actually `let NAME = EXPR1; EXPR2` and what it does is associates
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:112]:
```suggestion
full form is actually `let NAME = EXPR1; EXPR2` : what it actually does is associate
```

File: docs/tutorial/shell.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+the name `NAME` with the value of `EXPR1` and evaluates `EXPR2` with this
```
Comment (cuminandpaprika) [inline docs/tutorial/shell.md:113]:
```suggestion
the name `NAME` with the value of `EXPR1` and evaluate `EXPR2` with this
```

File: docs/tutorial/shell.md (hunk 6 above)
Comment (marcelocantos) [inline docs/tutorial/shell.md:112]:
```suggestion
full form is actually `let NAME = EXPR1; EXPR2`. What it does is associate
```

File: docs/tutorial/shell.md (hunk 3 above)
Comment (marcelocantos) [reply docs/tutorial/shell.md:95]: Agreed. After the docs are moved out of the main README (#186).

Comment (cuminandpaprika) [reply docs/tutorial/shell.md:95]: 👍

**Pull request merged**